npm/pod_test.go

```diff
@@ -0,0 +1,173 @@
+package npm
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-container-networking/npm/ipsm"
+
+	"github.com/Azure/azure-container-networking/npm/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestisValidPod(t *testing.T) {
+	podObj := &corev1.Pod{
+		Status: corev1.PodStatus{
+			Phase: "Running",
+			PodIP: "1.2.3.4",
+		},
+	}
+	if ok := isValidPod(podObj); !ok {
+		t.Errorf("TestisValidPod failed @ isValidPod")
+	}
+}
+
+func TestisSystemPod(t *testing.T) {
+	podObj := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: util.KubeSystemFlag,
+		},
+	}
+	if ok := isSystemPod(podObj); !ok {
+		t.Errorf("TestisSystemPod failed @ isSystemPod")
+	}
+}
+
+func TestAddPod(t *testing.T) {
+	npMgr := &NetworkPolicyManager{
+		nsMap: make(map[string]*namespace),
+	}
+
+	allNs, err := newNs(util.KubeAllNamespacesFlag)
+	if err != nil {
+		panic(err.Error)
+	}
+	npMgr.nsMap[util.KubeAllNamespacesFlag] = allNs
+
+	ipsMgr := ipsm.NewIpsetManager()
+	if err := ipsMgr.Save(util.IpsetTestConfigFile); err != nil {
+		t.Errorf("TestAddPod failed @ ipsMgr.Save")
+	}
+
+	defer func() {
+		if err := ipsMgr.Restore(util.IpsetTestConfigFile); err != nil {
+			t.Errorf("TestAddPod failed @ ipsMgr.Restore")
+		}
+	}()
+
+	podObj := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "test-pod",
+			Labels: map[string]string{
+				"app": "test-pod",
+			},
+		},
+		Status: corev1.PodStatus{
+			Phase: "Running",
+			PodIP: "1.2.3.4",
+		},
+	}
+	if err := npMgr.AddPod(podObj); err != nil {
+		t.Errorf("TestAddPod failed @ AddPod")
+	}
+}
+
+func TestUpdatePod(t *testing.T) {
```

Comment:
You can combine TestUpdatePod and TestAddPod (like TestAddUpdatePod) because 95% of the code is same. Is there any particular reason to have these separate? It's possible to combine delete in this as well. But I don't have any strong push for this